pkg/tlsx/openssl/openssl.go

```diff
@@ -44,7 +44,12 @@ func New(options *clients.Options) (*Client, error) {
 
 // Connect connects to a host and grabs the response data
 func (c *Client) ConnectWithOptions(hostname, ip, port string, options clients.ConnectOptions) (*clients.Response, error) {
-	address := net.JoinHostPort(hostname, port)
+	var address string
+	if ip != "" || c.options.ScanAllIPs || len(c.options.IPVersion) > 0 {
+		address = net.JoinHostPort(ip, port)
+	} else {
+		address = net.JoinHostPort(hostname, port)
+	}
 
 	if c.options.ScanAllIPs || len(c.options.IPVersion) > 0 {
 		address = net.JoinHostPort(ip, port)
```

Comment:
seems redundant already covered in L#48

Reply:
nice catch - forgot to delete it after merging conditions in the if